### PR TITLE
PMM-6457 Add API test for DBaaS

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-6457

ref:
- https://github.com/Percona-Lab/pmm-api-tests/pull/125
- https://github.com/percona-platform/dbaas-controller/pull/58
- https://github.com/percona/pmm-managed/pull/468
- https://github.com/percona/pmm/pull/582
- https://github.com/Percona-Lab/pmm-submodules/pull/1124

### CODE REVIEW

File: management/dbaas/xtra_db_cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+						ComputeResources: &xtra_db_cluster.CreateXtraDBClusterParamsBodyParamsPxcComputeResources{
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+				},
+			},
+		}
+
+		paramsFirstPXC.WithContext(pmmapitests.Context)
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:49]: Can we set Context while initializing the variable?

Comment (BupycHuk) [reply management/dbaas/xtra_db_cluster_test.go:49]: The same in all places.

File: management/dbaas/xtra_db_cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+					Name:                  pxc.Name,
+				},
+			}
+			deleteXtraDBClusterParamsParam.WithContext(pmmapitests.Context)
+			_, err := dbaasClient.Default.XtraDBCluster.DeleteXtraDBCluster(&deleteXtraDBClusterParamsParam)
+			assert.NoError(t, err)
+		}
+	})
+
+	t.Run("Create_XtraDBCluster_Invalid_Name", func(t *testing.T) {
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:143]:
```suggestion
	t.Run("CreateXtraDBClusterEmptyName", func(t *testing.T) {
```
Camel case and snake case in the same string?

File: management/dbaas/xtra_db_cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+						},
+					},
+				},
+			},
+		}
+		paramsPXCEmptyName.WithContext(pmmapitests.Context)
+		_, err := dbaasClient.Default.XtraDBCluster.CreateXtraDBCluster(&paramsPXCEmptyName)
+		pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `invalid field Name: value '' must not be an empty string`)
+
+		paramsPXCInvalidName := xtra_db_cluster.CreateXtraDBClusterParams{
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:169]: let's extract this part as seperate test

File: management/dbaas/xtra_db_cluster_test.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+							CPUm:        1,
+							MemoryBytes: "64",
+						},
+					},
+				},
+			},
+		}
+		paramsPXCInvalidName.WithContext(pmmapitests.Context)
+		_, err = dbaasClient.Default.XtraDBCluster.CreateXtraDBCluster(&paramsPXCInvalidName)
+		assert.Error(t, err)
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:192]: `pmmapitests.AssertAPIErrorf`?

File: management/dbaas/xtra_db_cluster_test.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+	t.Run("List_Unknown_Cluster", func(t *testing.T) {
+		listXtraDBClustersParamsParam := xtra_db_cluster.ListXtraDBClustersParams{
+			Body: xtra_db_cluster.ListXtraDBClustersBody{
+				KubernetesClusterName: "Unknown-kubernetes-cluster-name",
+			},
+		}
+		listXtraDBClustersParamsParam.WithContext(pmmapitests.Context)
+		_, err := dbaasClient.Default.XtraDBCluster.ListXtraDBClusters(&listXtraDBClustersParamsParam)
+		require.Error(t, err)
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:203]: `pmmapitests.AssertAPIErrorf`?

File: management/dbaas/xtra_db_cluster_test.go [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+	t.Run("Delete_Unknown_XtraDBCluster", func(t *testing.T) {
+		deleteXtraDBClusterParamsParam := xtra_db_cluster.DeleteXtraDBClusterParams{
+			Body: xtra_db_cluster.DeleteXtraDBClusterBody{
+				KubernetesClusterName: kubernetesClusterName,
+				Name:                  "Unknown-pxc-name",
+			},
+		}
+		deleteXtraDBClusterParamsParam.WithContext(pmmapitests.Context)
+		_, err := dbaasClient.Default.XtraDBCluster.DeleteXtraDBCluster(&deleteXtraDBClusterParamsParam)
+		require.Error(t, err)
```
Comment (BupycHuk) [inline management/dbaas/xtra_db_cluster_test.go:215]: `pmmapitests.AssertAPIErrorf`?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: management/dbaas/xtra_db_cluster_test.go (hunk 1 above)
Comment (askomorokhov) [reply management/dbaas/xtra_db_cluster_test.go]: Yes. Changed.

File: management/dbaas/xtra_db_cluster_test.go (hunk 2 above)
Comment (askomorokhov) [reply management/dbaas/xtra_db_cluster_test.go]: It's just the name of tests. There's no convention about it. We already have a zoo here. BTW - changed.

File: management/dbaas/xtra_db_cluster_test.go (hunk 3 above)
Comment (askomorokhov) [reply management/dbaas/xtra_db_cluster_test.go:170]: Agree. Done.

File: management/dbaas/xtra_db_cluster_test.go (hunk 4 above)
Comment (askomorokhov) [reply management/dbaas/xtra_db_cluster_test.go:193]: Not everywhere it is possible, as error contains a tmp random path to kubeconfig and depends on kubectl version/path.
I do not see big benefits to make any workaround here right now.

File: init.go [hunk 7]
```diff
@@ -176,6 +181,14 @@ func init() {
 		logrus.Fatalf("Failed to detect hostname: %s", err)
 	}
 
+	if *kubeconfigF != "" {
+		data, err := ioutil.ReadFile(*kubeconfigF)
+		if err != nil {
+			logrus.Fatalf("Failed to read kubeconfig: %s", err)
+		}
+		Kubeconfig = string(data)
```
Comment (percona-csalguero) [inline init.go:189]: The comment at line 52 says: `// Kubeconfig path to kubeconfig.` but we are storing the contents of the file here, as a string. The path is in the kingping flag `kubeconfigF`

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: init.go (hunk 7 above)
Comment (askomorokhov) [reply init.go:189]: Thanks. Fixed comment.

**Review by percona-csalguero — APPROVED** [review]

File: management/dbaas/xtra_db_cluster_test.go (hunk 4 above)
Comment (BupycHuk) [reply management/dbaas/xtra_db_cluster_test.go:193]: ok, can we check for a status code?

**Review by BupycHuk — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

**Review by artemgavrilov — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
But CI fails

**Pull request merged**